r/tests/testthat/test-dplyr-lubridate.R

```diff
@@ -26,22 +26,13 @@ library(dplyr)
 # TODO: consider reevaluating this workaround after ARROW-12980
 withr::local_timezone("UTC")
 
-test_date <- as.POSIXct("2017-01-01 00:00:12.3456789", tz = "")
+if (tolower(Sys.info()[["sysname"]]) == "windows") {
```

Comment:
Why is windows special-cased here? This needs an explanatory comment and possibly a link to a jira that will let us delete this.

Reply:
This is due to the [timezone db issue on windows](https://issues.apache.org/jira/browse/ARROW-13168). 
A comment here could state:
```
# TODO: We should test on windows once ARROW-13168 is resolved.
```

Reply:
If this doesn't work on windows, why did we have to add `-lole32` to r/configure.win in this patch?

Reply:
Without `-lole32` flag all zoned time logic was causing errors at compile time (if I remember correctly) so it would have to be disabled for windows. Since we will eventually have zones on windows I went ahead and rather added the flag.

Here's a [similar issue from date.h](https://github.com/HowardHinnant/date/issues/272) and a similar [discussion from another PR](https://github.com/apache/arrow/pull/10610#pullrequestreview-705668997).